bcipy/helpers/convert.py

```diff
@@ -28,65 +27,182 @@
-    edf_path - optional path to write converted data; defaults to writing
-        a file named raw.edf in the data_dir.
-    overwrite - If True, the destination file (if it exists) will be overwritten.
-        If False (default), an error will be raised if the file exists.
-    write_targetness - If True, and targetness information is available, write
-        that instead of the stimuli markers. False by default.
+    data_dir - directory which contains the data to be converted. This location must contain:
+        1. a parameter file.
+        2. a raw_data.csv file.
+        3. a trigger.txt file.
+    edf_path - optional; path to write converted data; defaults to writing a file named raw_data.edf in the data_dir.
+        Must end in .edf.
+    overwrite - If True, the destination file (if it exists) will be overwritten. If False (default), an error will
+        be raised if the file exists.
+    write_targetness - If True, and targetness information is available, write that instead of the stimuli markers.
+        False by default.
     use_event_durations - optional; if True assigns a duration to each event.
+    remove_pre_fixation - optional; if True removes the non-inquiry based markers from the trigger data.
+    pre_filter - optional; if True, apply the default filter to the data using to loaded parameters file.
 
     Returns
     -------
         Path to new edf file
     """
     if not edf_path:
-        edf_path = Path(data_dir, f'{RAW_DATA_FILENAME}.edf')
+        edf_path = str(Path(data_dir, f'{RAW_DATA_FILENAME}.edf').resolve())
+    if not edf_path.endswith('.edf'):
+        raise ValueError(f'edf_path=[{edf_path}] must end in .edf')
+
+    data, channels, fs, events, annotation_channels = pyedf_convert(
+        data_dir,
+        write_targetness=write_targetness,
+        use_event_durations=use_event_durations,
+        remove_pre_fixation=remove_pre_fixation,
+        pre_filter=pre_filter)
+
+    return write_edf(edf_path, data, channels, fs, events, overwrite, annotation_channels)
+
+
+def convert_to_bdf(data_dir: str,
+                   bdf_path: str = None,
+                   overwrite: bool = False,
+                   write_targetness: bool = False,
+                   use_event_durations: bool = False,
+                   remove_pre_fixation: bool = False,
+                   pre_filter: bool = False) -> Path:
+    """ Converts BciPy data to the BDF+ filetype using pyEDFlib.
+
+    See https://www.biosemi.com/faq/file_format.html for more detailed information about the BDF specification.
+    See https://www.teuniz.net/edfbrowser/ for a free EDF/BDF viewer.
+
+    Parameters
+    ----------
+    data_dir - directory which contains the data to be converted. This location must contain:
+        1. a parameter file.
+        2. a raw_data.csv file.
+        3. a trigger.txt file.
+    edf_path - optional; path to write converted data; defaults to writing a file named raw_data.edf in the data_dir.
```

Comment:
The documentation should be updated to use`bdf_path` and the file name should end in `.bdf`.